### ISSUE TITLE
F3X BTools not working in new places

### ISSUE THREAD

**Issue by zertyya** [F3XTeam/RBX-Building-Tools#163]
Since at least  the 31/12/23, F3X appears to be broken in newly created projects, both in-game and in studio. Before beginning, the studio plugin works fine. Just the tool itself seems to be quite broken.

In both studio and in-game when you equip the tool this error occurs:

`Players.Player.Backpack.Building Tools.Libraries.Signal:54: Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.`

When resumed and the tools use is attempted in any form, the following error occurs:

`Players.Player.Backpack.Building Tools.Libraries.Signal:46: invalid argument #1 to 'unpack' (table expected, got nil)`

While testing in-game I found the following errors within a few minutes of experimentation:
>When selecting the different tools that none of them highlighted in the GUI menu
>When switching from the move tool and then to the resize tool and deselecting the part, handles didn't disappear 
>Occasionally the highlight around selected parts would stick when deselected
>Copying parts deselected the newly generated parts
>When switching selection of a part sometimes handles would stick to the last part
>When parts are deleted the handles are not removed

Images can be seen below to help describe my issues as I'm not great at writing descriptions. Quick experimentation will be able to get across what I'm saying.

Older games seem to work fine. Seems to be newly created games that are buggy.

![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/157382197/46853d56-94ba-4641-a6ee-79d064ac53d4)
![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/157382197/e6fcb0c6-778d-40cd-9aa3-c19ec1fce351)
![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/157382197/c99720cd-0a9a-4625-9f3a-f8fac2531761)
![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/157382197/f01c0610-d186-4d0a-9e67-00d943a2c09d)
![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/157382197/54cdb3e6-9aa4-41bd-8fb3-4109c83ee94b)

**Comment by FrutigerMetro** [comment]
for some reason new places have SignalBehaviour set to Deferred in workspace properties, change it to Immediate and it should work as normal!
![image](https://github.com/F3XTeam/RBX-Building-Tools/assets/73192529/0fb29106-3c8c-42be-a1e5-1ae03e047486)

**Comment by ThatNemusy** [comment]
@TonyRBXM 

Also, the part creation doesn't close and move to move tool after part are created in F3X.
How would you fix that?

**Comment by ThatNemusy** [comment]
> Also, the part creation doesn't close and move to move tool after part are created in F3X. How would you fix that?

@twixhood

**Comment by FrutigerMetro** [comment]
No clue, it seems to me that F3X is kinda broken right now. It doesn’t work correctly on my game again for some reason.

**Comment by ThatNemusy** [comment]
I mean, on toolbox

The f3x has been officially fixed

Thank you for trying to help tho
________________________________

**Comment by FrutigerMetro** [comment]
It’s been fixed? Finally

**Comment by ThatNemusy** [comment]
Yeah, but you still have to set signalbehaviour to immediate, i think

________________________________
